astropy/io/votable/tests/converter_test.py

```diff
@@ -146,6 +146,41 @@ def test_float_mask_permissive():
     assert c.parse('null', config=config) == (c.null, True)
 
 
+def test_double_array():
+    config = {'verify': 'exception'}
+    field = tree.Field(None, name='c', datatype='double', arraysize='3',
+                       config=config)
+    data = (1.0, 2.0, 3.0)
+    c = converters.get_converter(field, config=config)
+    assert c.output(1.0, False) == '1'
+    assert c.output(1.0, [False, False]) == '1 1'
+    assert c.output(data, False) == '1 2 3'
+    assert c.output(data, [False, False, False]) == '1 2 3'
+    assert c.output(data, [False, False, True]) == '1 2 NaN'
+
+    with pytest.raises(ValueError, match='shape mismatch'):
+        c.output(data, [False, False])
+
+    a = c.parse("1 2 3", config=config)
+    assert_array_equal(a[0], data)
+    assert_array_equal(a[1], False)
+
+    with pytest.raises(exceptions.E02):
+        c.parse("1", config=config)
+
+    with pytest.raises(AttributeError):
```

Comment:
This is probably okay to error out like this rather than emitting `E02` as warning and then assume things for the output?